src/openassetio-python/tests/package/pluginSystem/test_hybridpluginsystemmanagerimplementationfactory.py

```diff
@@ -0,0 +1,1637 @@
+        with pytest.raises(
+            errors.InputValidationException, match="Manager implementation factory cannot be None"
+        ):
+            HybridPluginSystemManagerImplementationFactory([factory_a, None], mock_logger)
+
+
+class Test_HybridPluginSystemManagerImplementationFactory_identifiers:
+    def test_when_no_child_factories_then_no_identifiers(self, mock_logger):
+        factory = HybridPluginSystemManagerImplementationFactory([], mock_logger)
+        assert factory.identifiers() == []
+
+    def test_when_child_factories_then_identifiers_from_all_children_deduplicated(
+        self, mock_logger, factory_a, factory_b
+    ):
+        factory_a.mock.identifiers.return_value = ["a", "c", "b"]
+        factory_b.mock.identifiers.return_value = ["b", "c", "d"]
+
+        factory = HybridPluginSystemManagerImplementationFactory(
+            [factory_b, factory_a], mock_logger
+        )
+        assert factory.identifiers() == ["a", "b", "c", "d"]
+
+
+class Test_HybridPluginSystemManagerImplementationFactory_instantiate:
+    def test_when_no_child_factories_then_error_raised(self, mock_logger):
+        factory = HybridPluginSystemManagerImplementationFactory([], mock_logger)
+        with pytest.raises(
+            errors.InputValidationException,
+            match="HybridPluginSystem: No plug-in registered with the identifier 'foo'",
+        ):
+            factory.instantiate("foo")
+
+    def test_when_no_match_from_multiple_child_factories_then_error_raised(
+        self, factory_a, factory_b, mock_logger
+    ):
+        factory_a.mock.identifiers.return_value = ["bar"]
+        factory_b.mock.identifiers.return_value = ["baz"]
+
+        factory = HybridPluginSystemManagerImplementationFactory(
+            [factory_a, factory_b], mock_logger
+        )
+
+        with pytest.raises(
+            errors.InputValidationException,
+            match="HybridPluginSystem: No plug-in registered with the identifier 'foo'",
+        ):
+            factory.instantiate("foo")
+
+    def test_when_single_factory_matches_then_uses_that_factorys_interface(
+        self, mock_logger, factory_a, factory_b, the_plugin_identifier
+    ):
+        factory_a.mock.identifiers.return_value = ["a", the_plugin_identifier]
+        factory_b.mock.identifiers.return_value = ["b"]
+
+        factory = HybridPluginSystemManagerImplementationFactory(
+            [factory_a, factory_b], mock_logger
+        )
+
+        actual_manager_interface = factory.instantiate(the_plugin_identifier)
+
+        factory_a.mock.instantiate.assert_called_once_with(the_plugin_identifier)
+        assert not factory_b.mock.instantiate.called
+        assert actual_manager_interface is factory_a.mock.instantiate.return_value
+
+    def test_when_many_factories_provided_and_one_matches_then_uses_that_factorys_interface(
+        self,
+        mock_logger,
+        create_mock_manager_interface,
+        factory_a,
```

Comment:
That's interesting.  Some time ago we did discuss as a team whether utility code should come before or after its usage.  

I successfully argued for the latter, reasoning that the high-level code should always be the first thing you read from top to bottom, since it provides the context and is usually what you're interested in.

Perhaps test fixtures, and tests in general, are a special case, though?

Reply:
It's how I ask recipes (thus Python) be written, that utility methods are local to their first use, and appear before their first use. It's because I read recipes top-down.

Reply:
Resolving this, as you already had this layout agreed upon.